legal-api/tests/unit/core/test_filing.py

```diff
@@ -189,10 +191,24 @@ def test_set_effective(session):
         filing._storage.payment_completion_date = alt_payment_date
         filing._storage.save()
 
-        filing.storage.set_processed()
+        filing.storage.set_processed(legal_type)
 
         # assert that the effective date is the future date
         assert filing._storage.effective_date
         assert filing._storage.effective_date.replace(tzinfo=None) == future_date.replace(tzinfo=None)
         assert filing.is_future_effective
 
+        past_date = now - datedelta.DAY
+        filing.storage.effective_date = past_date
+        filing._storage.skip_status_listener = True
+        filing._storage.payment_completion_date = payment_date
+        filing._storage.save()
+
+        legal_type = 'CP'
+        filing.storage.set_processed(legal_type)
+
+        # assert that the effective date is unchanged by payment date
+        assert filing._storage.effective_date
+        assert filing._storage.effective_date.replace(tzinfo=None) != payment_date.replace(tzinfo=None)
```

Comment:
can we also add an assertion that the effective date == past_date?

Reply:
Yes I'll add that.